common/lib/dependabot/git_commit_checker.rb

```diff
@@ -115,6 +115,16 @@ def local_tag_for_latest_version
     end
     # rubocop:enable Metrics/PerceivedComplexity
 
+    def local_tag_for_pinned_version
+      return unless pinned?
+
+      ref = dependency_source_details.fetch(:ref)
+      tags = local_tags.select { |t| t.commit_sha == ref }
+      return if tags.empty?
+
+      tags.first.name
```

Comment:
This probably needs a sort first - e.g. if a single commit is tagged with `v2.3.4`, `v2.3` and `v2`.

That's a common pattern.